runtime/vm/jvminit.c

```diff
@@ -2432,8 +2432,8 @@ IDATA VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved) {
 			TRIGGER_J9HOOK_VM_ABOUT_TO_BOOTSTRAP(vm->hookInterface, vm->mainThread);
 			/* At this point, the decision about which interpreter to use has been made */
 			vm->bytecodeLoop = J9_ARE_ANY_BITS_SET(vm->extendedRuntimeFlags, J9_EXTENDED_RUNTIME_DEBUG_MODE)
-					? (void*)debugBytecodeLoop
-					: (void*)bytecodeLoop;
+					? debugBytecodeLoop
```

Comment:
This will likely fail to compile on other platforms. Not clear why this change is necessary.

Reply:
@gacholio ,  the change here is the key to call the interpreter in /runtime/vm/rv64cinterpnojit.cpp
```
void
cInterpreter(J9VMThread *currentThread)
{
	J9JavaVM *vm = currentThread->javaVM;
	vm->bytecodeLoop(currentThread); <------
}
```
Otherwise, the VM will totally fail to start.  Technically it might be correct to directly call bytecodeLoop() here like by modi:
```
void
cInterpreter(J9VMThread *currentThread)
{
     bytecodeLoop(currentThread); <------
}
``
But I am not too sure whether there is any issue introduced with this approach.

Reply:
I still don't see why you need to remove the cast. And no, you can't call it directly - it has to be called indirect to support debug (and the upcoming mixed mode).

Reply:
I still don't see why you need to remove the cast. <---- this change was proposed previously by @keithc-ca  as follows:
```
According to vm_api.h [1] those functions have signature UDATA (*)(J9VMThread *) while the assignment in jvminit.c expects this field to be of type void * [2]: I don't expect this to compile without further changes.

[1] https://github.com/eclipse/openj9/blob/master/runtime/oti/vm_api.h#L4069-L4080
[2] https://github.com/eclipse/openj9/blob/master/runtime/vm/jvminit.c#L2414-L2416
```

Reply:
Sorry, didn't see that you had also changed the type in the VM structure. I agree this is more correct, otherwise you'd have to do the cast before the call.